packages/renderer/src/lib/deployments/DeploymentUI.ts

```diff
@@ -18,9 +18,17 @@
 
 export interface DeploymentUI {
   name: string;
+  status: string;
   namespace: string;
   replicas: number;
   ready: number;
+  created: Date | undefined;
   age: string;
   selected: boolean;
+  conditions: DeploymentCondition[];
+}
+
+export interface DeploymentCondition {
+  type: string;
+  message: string | undefined;
```

Comment:
I think you need to review other places where we need to use the optional `?` vs `| undefined`
```
message? : string
```